### PR TITLE
Add workflow for building Docker image on commit to main

### PR DESCRIPTION
**Pull request by vegarsti**
Postponing pushing to a public registry, need to decide on the offical name.

### CODE REVIEW

**Comment by vegarsti** [comment]
* **#4** <a href="https://app.graphite.dev/github/pr/duneanalytics/blockchain-ingester/4?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vegarsti and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+COPY . /app/
+RUN make build
+
+FROM debian:bookworm-slim
+RUN apt update \
+	&& apt install -y ca-certificates \
+	&& apt clean \
+	&& rm -rf /var/lib/apt/lists/*
+
+COPY --from=builder /app/dist/ /app/
```
Comment (msf) [inline Dockerfile:23]: copy /app/ingester

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ENTRYPOINT ["/app/duneapi"]
```
Comment (msf) [inline Dockerfile:24]: /app/ingester

File: Makefile [hunk 3]
```diff
@@ -16,7 +16,7 @@ bin/golangci-lint:
 bin/gofumpt: bin
 	GOBIN=$(PWD)/bin go install mvdan.cc/gofumpt@v0.6.0
 
-harvester: lint cmd/main.go
+build: lint cmd/main.go
 	go build -o harvester cmd/main.go
```
Comment (msf) [inline Makefile:20]:
```suggestion
	go build -o ingester cmd/main.go
```

**Review by msf — COMMENTED** [review]
we just need to copy the binary: ingester

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,24 @@
+FROM golang:1.22-bookworm AS builder
```
Comment (msf) [inline Dockerfile:1]: I wonder if we should use `1.22-alpine`

**Comment by vegarsti** [comment]
Regarding base Docker images and debian etc. I'm confused whether or not you're suggesting we change. I don't know enough about alpine/debian to know what would be best

**Comment by vegarsti** [comment]
Ran successfully in the PR, changing it to run on merge to main

File: .github/workflows/pr.yaml [hunk 5]
```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   test:
-    name: "Build Docker image and push to Dockerhub"
+    name: "Lint and test"
```
Comment (vegarsti) [inline .github/workflows/pr.yaml:11]: I messed up the name in https://github.com/duneanalytics/blockchain-ingester/pull/3I messed up the name in https://github.com/duneanalytics/blockchain-ingester/pull/3

**Pull request merged**